code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/models/CarouselPushTemplate.kt

```diff
@@ -112,6 +114,14 @@ internal open class CarouselPushTemplate : AEPPushTemplate {
         private const val SELF_TAG = "CarouselPushTemplate"
         const val MINIMUM_FILMSTRIP_SIZE = 3
 
+        fun createCarouselPushTemplate(data: Map<String, String>): CarouselPushTemplate {
+            val carouselPushTemplate = CarouselPushTemplate(data)
```

Comment:
Instead of constructing the entire object can you just read this value? 
val carouselLayoutType = DataReader.optString(
            data, PushTemplateConstants.PushPayloadKeys.CAROUSEL_LAYOUT, null
 )